### PR TITLE
[avrdude] Add recipe

### PR DESCRIPTION
**Pull request by Seelengrab**

### CODE REVIEW

File: A/avrdude/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl")
+]
```
Comment (giordano) [inline A/avrdude/build_tarballs.jl:35]:
```suggestion
platforms = supported_platforms(; exclude=!Sys.islinux)
```

File: A/avrdude/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/avrdudes/avrdude.git", "2e0be1e1aedfc10c904712bddda37e36356e7a66")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/avrdude/
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release
+make
```
Comment (giordano) [inline A/avrdude/build_tarballs.jl:17]:
```suggestion
make -j${nproc}
```

Comment (Seelengrab) [reply A/avrdude/build_tarballs.jl:17]: The build of this is so fast, it hardly seemed worth it - it took just 19s on CI and 1m total. The vast majority of that seems to be the download.

File: A/avrdude/build_tarballs.jl (hunk 1 above)
Comment (Seelengrab) [reply A/avrdude/build_tarballs.jl:35]: Good idea - that long list of stuff is just what the wizard spit out :)

**Pull request merged**